### ISSUE TITLE
Issue when using -webkit-text-stroke

### ISSUE THREAD

**Issue by PolMrt** [rsms/inter#292]
**Describe the bug**
When using css class `-webkit-text-stroke-color` & `-webkit-text-stroke-width`to put a border fitting the text, there are some unwantend border. Like this : 
<img width="356" alt="" src="https://user-images.githubusercontent.com/28253868/85221188-1cfb4000-b3b2-11ea-8054-8745e2f17f93.png">

**To Reproduce**
1. Create a `<div>`
2. Put some text in it, especialy these characters : `t`, `k`, `&`, `e`
3. Add these css classes : `-webkit-text-stroke-color: red; -webkit-text-stroke-width: .8px; color: transparent` (it also works when setting a color to the text)

**Expected behavior**
This should looks like this (here, using system font) : 
<img width="350" alt="Capture d’écran 2020-06-21 à 11 26 04" src="https://user-images.githubusercontent.com/28253868/85221250-98f58800-b3b2-11ea-88d3-6ca4873ac2ed.png">

**Environment**
- OS: macOS 10.14
- App that renders the font : firefox
- Version of font : 3.13

**Comment by thagxt** [comment]
I came here for the same reason! +1

**Comment by rsms** [comment]
Did you by any chance use the fonts from Google Fonts?
If you did, can you please try with the official release from https://github.com/rsms/inter/releases/latest ?

**Comment by thagxt** [comment]
I also use the CDN: https://rsms.me/inter/inter.css

**Comment by rsms** [comment]
I see! You are using the variable font.

TL;DR: Use static/constant fonts instead of the variable version.

<img width="432" alt="Screen Shot 2020-08-17 at 09 47 43" src="https://user-images.githubusercontent.com/47046/90422083-2ed62800-e06f-11ea-8c58-fa463990987f.png">

This is unfortunately a bug (or really, a limitation) of the rendering engine in some web browsers. The variable font uses overlapping shapes to be able to be variable (technical reasons.) Most quality font renderers will apply what is usually called "boolean union" on the shapes of a glyph prior to rendering. Not doing this causes issues with regular "filled" rendering in addition to stroked rendering. 

For example, Figma correctly applies the boolean union on the glyph's shapes _before_ generating stroke geometry:
<img width="753" alt="Screen Shot 2020-08-17 at 09 52 04" src="https://user-images.githubusercontent.com/47046/90422254-752b8700-e06f-11ea-987e-7bb3af31bdce.png">

However, some design tools like Sketch fails, just like some web browsers:
<img width="672" alt="Screen Shot 2020-08-17 at 09 52 30" src="https://user-images.githubusercontent.com/47046/90422281-807eb280-e06f-11ea-9875-69a428029735.png">

So, this is unfortunately an issue with the software rendering the font.

**Comment by apodtele** [comment]
Would you know how "boolean union" is performed on outlines/contours? I bet it is not trivial.

**Comment by rsms** [comment]
https://en.wikipedia.org/wiki/Boolean_operations_on_polygons

**Comment by apodtele** [comment]
The described algorithms operate on bitmaps or scan sweeps, i.e. during rendering not prior to it. It is rather expensive computationally. I was hoping that there was a good algorithm to merge outlines properly, which would be appropriate for this bug. Also, the overlap of the stroke paths is technically minute here unless you really track the winding number is a very special way.

To summarize, it is not a trivial problem and the overlaps were accepted in variable fonts rather hastily.
